### ISSUE TITLE
Replace types.NoneType with None in docstring

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8456]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8456|
|Type|enhancement|
|Created|2016-06-12 04:37:30Z|
|Branch|https://github.com/twisted/twisted/tree/8456-rodrigc-nonetype|

The *types* module constants have been removed in Python 3.
For example using *types.TupleType* in Python 3 will give an *AttributeError*

These constants should be removed and replaced with the primitive type instead.  See:

http://www.diveintopython3.net/porting-code-to-python-3-with-2to3.html#types

<details><summary>Searchable metadata</summary>

```
trac-id__8456 8456
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8456_rodrigc_nonetype 8456-rodrigc-nonetype
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1465706250051309 1465706250051309
changetime__1466574778921467 1466574778921467
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/175

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

The preferred marker for None is just `C{None}` as everybody should know what None is and what is its type.

Please update the current comments, or better aggregate all comment updates into a single PR :)

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @adiroiban|
|-|-|

See:
https://github.com/twisted/twisted/pull/195

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban removed owner|
|-|-|

I have pushed a branch and updated C{None} to L{None} everywhere.

I have also updated the documentation to make it clear what should be the markup for the stdlib types.

Since I have also made changes to this branch, it needs to be reviewed by someone else.

The diff is big but is just documentation changes + C{None} to L{None} everywhere.

I feel that submitting a single patch for this change is better than breaking into 10 tickets... some of them only having a few lines and each ticket doing the same change. 

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set owner to @adiroiban|
|-|-|

Looks OK.  Ready to land.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 8c25f2c86652276474dfdc8fee70f159369b1d13
```
#!CommitTicketReference repository="" revision="8c25f2c86652276474dfdc8fee70f159369b1d13"
Merge 8456-travis-pyflakes: Use L{None} markup in all docstrings.

Author: adiroiban
Reviewer: rodrigc
Fixes: #8456
```
